test/cpp/api/modules.cpp

```diff
@@ -563,7 +563,7 @@ TEST_F(ModulesTest, Unflatten) {
     return dimnames;
   };
   
-  unflatten = Unflatten(UnflattenOptions("B", {{"B1", 2}, {"B2", 2}}));
+  unflatten = Unflatten(UnflattenOptions("B", UnflattenOptions::namedshape_t{{"B1", 2}, {"B2", 2}}));
```

Comment:
Sry, I think I was not clear. This still won't work since you are still using initialize_list to initialize tuple (or pair). The initialize_list can not deduce the types within tuple (or pair).
Try explicit declare the type of tuple within initialize_list, something lie { tuple<X, Y>{"B1", 2}, tuple<X, Y>("B2", 2)}. I know this is not nice, but no other way